subsys/net/lib/download_client/src/transports/http.c

```diff
@@ -0,0 +1,575 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
```

Comment:
I did more tests with P-GPS prediction download and I no longer saw the failure at the beginning of the download. However, the download stalled ~200 bytes before the end and seems to be stuck after that. Ping works, but there hasn't been any events from the Download client anymore. So far I have seen this only once, I'll do some more runs.
```
[00:00:23.928,283] <inf> nrf_cloud_pgps: Storing prediction num:40 idx:40 for gps sec:1415671200
[00:00:23.928,649] <inf> download_client: Downloaded 82555/84268 bytes (97%)
[00:00:24.068,664] <inf> download_client: Downloaded 83933/84268 bytes (99%)
[00:00:24.068,939] <inf> download_client: Downloaded 84056/84268 bytes (99%)
RRC mode: Idle
mosh:~$ ping -d 8.8.8.8
Initiating ping to: 8.8.8.8
Source IP addr: 10.128.2.197
Destination IP addr: 8.8.8.8
Modem domain event: CE-level 0
RRC mode: Connected
Pinging 8.8.8.8 results: time=0.298secs, payload sent: 0, payload received 0
Pinging 8.8.8.8 results: time=0.062secs, payload sent: 0, payload received 0
Pinging 8.8.8.8 results: time=0.051secs, payload sent: 0, payload received 0
Pinging 8.8.8.8 results: time=0.053secs, payload sent: 0, payload received 0
Ping statistics for 8.8.8.8:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss)
Approximate round trip times in milli-seconds:
    Minimum = 51ms, Maximum = 298ms, Average = 116ms
Pinging DONE
RRC mode: Idle
mosh:~$
```

Reply:
I did more testing, but couldn't reproduce this. Marking as resolved.